cmd/prometheus/main_test.go

```diff
@@ -31,35 +30,23 @@ import (
 	"github.com/prometheus/prometheus/util/testutil"
 )
 
-var promPath string
+var promPath = os.Args[0]
 var promConfig = filepath.Join("..", "..", "documentation", "examples", "prometheus.yml")
 var promData = filepath.Join(os.TempDir(), "data")
 
 func TestMain(m *testing.M) {
-	flag.Parse()
-	if testing.Short() {
-		os.Exit(m.Run())
+	for i, arg := range os.Args {
+		if arg == "-test.main" {
+			os.Args = append(os.Args[:i], os.Args[i+1:]...)
```

Comment:
Note that I have check that it does not panic even if `-test.main` is last:
https://play.golang.org/p/IT0mKhfStGh